### PR TITLE
Modify API response URLs for organization specific routing scenarios

### PR DESCRIPTION
**Pull request by dewniMW**
Issue: https://github.com/wso2/product-is/issues/14060

Depends on: https://github.com/wso2/carbon-kernel/pull/3358

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -525,7 +531,7 @@
         <identity.inbound.saml2.version>5.8.44</identity.inbound.saml2.version>
         <commons.beanutils.version>1.9.4</commons.beanutils.version>
         <mavan.findbugsplugin.exclude.file>findbugs-exclude-filter.xml</mavan.findbugsplugin.exclude.file>
-        <carbon.kernel.version>4.7.0-m1</carbon.kernel.version>
+        <carbon.kernel.version>4.7.0-beta8</carbon.kernel.version>
```
Comment (dewniMW) [inline pom.xml:534]: need to bump version

File: components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java [hunk 2]
```diff
@@ -68,8 +70,18 @@ public static String getUsernameFromContext() {
...
+     * @return the organization id.
+     */
+    public static String getOrganizationIdFromContext() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
+    }
+
+    /**
+     * Build URI prepending the server API context with the proxy context path to the endpoint.
+     * Ex: /t/<tenant-domain>/api/server/<endpoint> or /o/<organization-domain>/api/server/<endpoint>
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java:84]:
```suggestion
     * Ex: /t/<tenant-domain>/api/server/<endpoint> or /o/<organization-id>/api/server/<endpoint>
```

File: components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java [hunk 3]
```diff
@@ -89,8 +101,9 @@ public static URI buildURIForBody(String endpoint) {
     }
 
     /**
-     * Build the complete URI prepending the user API context without the proxy context path, to the endpoint.
-     * Ex: https://localhost:9443/t/<tenant-domain>/api/users/<endpoint>
+     * Build the complete URI prepending the server API context without the proxy context path, to the endpoint.
+     * Ex: https://localhost:9443/t/<tenant-domain>/api/server/<endpoint> or
+     *     https://localhost:9443/o/<organization-domain>/api/server/<endpoint>
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java:106]:
```suggestion
     *     https://localhost:9443/o/<organization-id>/api/server/<endpoint>
```

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java (hunk 2 above)
Comment (dewniMW) [reply components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java]: fixed

File: components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java (hunk 3 above)
Comment (dewniMW) [reply components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java]: fixed

**Pull request merged**